src/main/java/stream/flarebot/flarebot/util/ConfirmLib.java

```diff
@@ -0,0 +1,66 @@
+package stream.flarebot.flarebot.util;
+
+import net.dv8tion.jda.core.requests.RestAction;
+import stream.flarebot.flarebot.commands.Command;
+import stream.flarebot.flarebot.commands.moderation.PruneCommand;
+import stream.flarebot.flarebot.objects.RestActionWrapper;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+public class ConfirmLib {
+
+    private static final ExpiringMap<String, List<RestActionWrapper>> confirmMap = new ExpiringMap<>(TimeUnit.MINUTES.toMillis(1));
+
+    public static void pushAction(String userID, RestActionWrapper action) {
+        if (confirmMap.containsKey(userID)) {
+            List<RestActionWrapper> actions = confirmMap.get(userID);
+            if (actions.stream().filter(wrapper -> wrapper.getOrigin().equals(action.getOrigin())).count() == 0) {
+                actions.add(action);
+            }
+        } else {
+            ArrayList<RestActionWrapper> actions = new ArrayList<>();
```

Comment:
Use a concurrent hash set (`ConcurrentHashMap.newKeySet()`) for thread safety and speed and duplication checking